src/Sdk/WebApi/WebApi/ResultsHttpClient.cs

```diff
@@ -214,9 +214,15 @@ private async Task UploadBlockFileAsync(string url, string blobStorageType, File
             if (m_useSdk && blobStorageType == BlobStorageTypes.AzureBlobStorage)
             {
                 var blobClient = GetBlobClient(url);
+                var uploadOptions = new BlobUploadOptions();
+                string extension = System.IO.Path.GetExtension(file.Name);
+                if (extension == ".txt")
+                {
+                    uploadOptions.HttpHeaders.ContentType = "text/plain";
+                }
```

Comment:
> @bethanyj28 How are we handling blob downloads that actually should have a content type of `application/octet-stream`? For example, we want this for downloading raw logs.

@jtamsut this is the way we chatted about earlier, let me know if you have something else in mind.

Reply:
Ah I missed this. Okay this looks good to me!